torch/nn/modules/module.py

```diff
@@ -1016,8 +1016,43 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                     if input_name not in self._modules and input_name not in local_state:
                         unexpected_keys.append(key)
 
+
+    def _strip_prefix_from_state_dict_if_exists(self, state_dict, prefix: str):
+        r"""This function strips a prefix if it exists in the model
+        state_dict and metadata.
+
+        Arguments:
+            state_dict (dict) : DP/DDP pytorch model state_dict
+            prefix (str) : Prefix to be removed from the DP/DDP model state_dict to
+                           to make it compatible with regular pytorch model.
+        Returns:
+            state_dict (dict)
+        """
+        keys = sorted(state_dict.keys())
+        if not all(len(key) == 0 or key.startswith(prefix) for key in keys):
+            return
+
+        for key in list(state_dict.keys()):
+            new_key = key[len(prefix):]
+            state_dict[new_key] = state_dict.pop(key)
+
+            metadata = state_dict._metadata
+
+        else:
+            for key in list(metadata.keys()):
+                # for the metadata dict, the key can be:
+                # '': for the DDP module, which we want to remove.
+                # 'module': for the actual model.
+                # 'module.xx.xx': for the rest.
+
+                if len(key) == 0:
+                    continue
+                newkey = key[len(prefix):]
+                metadata[newkey] = metadata.pop(key)
+        return state_dict
+
     def load_state_dict(self, state_dict: Union[Dict[str, Tensor], Dict[str, Tensor]],
-                        strict: bool = True):
+                        strict: bool = True, is_parallel: bool = False):
```

Comment:
Hey @ganprad  We will need an RFC issue for this API change. Is there any discussion that we can refer to for this change?

Also loop in PT Frontend owners cc @gchanan @ezyang

Reply:
From the distributed package side, I am not sure which one is better: let it fail with clear error message vs make it implicitly work. If users would like to load the local module, can they save `ddp.module` instead? So that both save and load can be explicit.

Reply:
Can you explain why you need the user to say so here? Why can't `load_state_dict` automatically detect the situation and do the right thing?